### ISSUE TITLE
Archive search accessibility issues 

### ISSUE THREAD

**Issue by rlskoeser** [Princeton-CDH/ppa-django#238]
According to WAVE: 
- [x] 3 empty links (pagination, back to top ...?) - error
- [x] 15 orphaned form labels (collections, publication date ? ) - warning
- [x]  2 Unlabeled form element with title (publication date) - warning
- [x] search results flagged as possible headings; can we use h2 instead of spans flagged as headers?

**Comment by thatbudakguy** [comment]
not seeing any of these anymore via axe except the unlabeled pub date inputs, which should be fixed now.
